File: Java/BinarySearch.java
public class BinarySearch {

  public static void main(String args[]) {  
        int arr[] = {10,20,30,40,50};  
        int key = 30;  
        int last = (arr.length) - 1;  
        
        int result = binarySearch(arr, 0, last, key);  
        
        if (result == -1)  
            System.out.println("Element is not found!");  
        else  
            System.out.println("Element is found at index: " + result);  
    }
  
  public static int binarySearch(int arr[], int first, int last, int key){  
        if (last >= first){  
            int mid = first + (last - first) / 2;  
            
            if (arr[mid] == key){  
              return mid;  
            } 
            
            if (arr[mid] > key){  
              return binarySearch(arr, first, mid-1, key);
            
            }else{  
              return binarySearch(arr, mid+1, last, key);
            }  
        } 
        return -1;  
    }  

}

File: C++/binary_Search.cpp
#include<iostream>
using namespace std;

int binarySearch(int arr[],int n,int key)
{
    int s=0;
    int e=n;
    while(s<e)
    {
        int mid=s+e/2;

        if(arr[mid]==key)
        {
            return mid;
        }
        else if(arr[mid]>key)
        {
            e=e-1;
        }
        else
        {
            s=s+1;
        }
    }
    return -1;
}


int main()
{
    int n;
    int key;
    int arr[n];

     cin>>n;
       
    for(int i=0;i<n;i++)
    {
        cin>>arr[i];
    }
    cout<<"Enter the key"<<ends;
    cin>>key;
    cout<<binarySearch(arr,n,key);

    return 0;
}